src/patterns/strategy/context.impl.ts

```diff
@@ -0,0 +1,13 @@
+import {IBaseDiscountStrategy, IStrategyContext} from "./interfaces";
+
+export class StrategyContext implements IStrategyContext{
```

Comment:
In your variant of StrategyContext it looks like the class is the redundant proxy, because you can just call `strategy.calculateDiscount`.
In strategy pattern the context allows to dynamically switch strategies in runtime, so it should have getter and setter